com.reprezen.swagedit/src/com/reprezen/swagedit/editor/DocumentUtils.java

```diff
@@ -74,14 +74,7 @@ public static IFile getWorkspaceFile(URI uri) {
 		IWorkspaceRoot root = ResourcesPlugin
 				.getWorkspace()
 				.getRoot();
-
-		URI relativize = root.getLocationURI().relativize(uri);
-		IPath path = new Path(relativize.getPath());
-		try {
-			return root.getFile(path);
-		} catch (Exception e) {
-			return null;
-		}
+		return root.getFileForLocation(new Path(uri.getPath()));
 	}
```

Comment:
Documentation for the [`IWorkspaceRoot#getFileForLocation`](http://help.eclipse.org/mars/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Fresources%2FIWorkspaceRoot.html) says, " The path should be absolute; a relative path will be treated as absolute."

The previous code seems to anticipate an absolute path, thus the call to `relativize`.  But the new implementation appears to _require_ an absolute URI.  We should document that precondition in the Javadoc and maybe check to make sure it is met; or else coerce to an absolute path.